packages/@aws-cdk/aws-servicecatalog/README.md

```diff
@@ -184,6 +184,48 @@ const product = new servicecatalog.CloudFormationProduct(this, 'Product', {
 });
 ```
 
+We can also define a `versioningStrategy` for your `ProductStack` deployment. 
+With the `Default` strategy, a new `ProductStack` is needed for each `productVersion`
+and each `productVersion` will get overwritten with the latest changes to your `ProductStack`.
+With the `RetainPreviousVersions` strategy, a local template will be retained for each `ProductStack` deployed.
+Subsequently, any `ProductStack` changes will not overwrite the current version.
+A new `productVersionName` must be specified in order for `ProductStack` changes to be deployed.
+When the `ProductStack` is updated with a new version, the previous version can still be deployed.
+The template of previously deployed `ProductStack` can be referenced using `fromProductStackContext`
+and passing the corresponding `ProductStack` id.
```

Comment:
This might be a sign of poor naming.

Reply:
If a user specifies the `Default` strategy then switches to `RetainPreviousVersions`. Only versions deployed after the switch will be retained. 
If they do the reverse, then all further deployed versions will no longer be retained. Previously retained versions will not be deleted.

Do you have a suggestion on how we can name this behavior better?

Reply:
So `RetainPreviousVersions` and `Default` only apply to the version on which they are attached. I find this confusing because we have the concept of a Version referring to both a version of a product and a version of a version. I understand it's up to the user to attach semantics to these terms, but I'd prefer to see versions of a product called variants (or something better). @rix0rrr thoughts?

Reply:
Are you not trying to say something like:

```
const product = new CloudFormationProduct(this, 'Product', {
  productName: "My Product",
  owner: "Product Owner",
  productVersions: [
    {
      productVersionName: "v1",
      cloudFormationTemplate: CloudFormationTemplate.fromSnapshot('v1'),
    },
  ],
});
```

?